### DISCUSSION TITLE
Build problems

### DISCUSSION THREAD

**Discussion by dbosnyak** [alf45tar/PedalinoMini#331] [Q&A] [answered]
Hi,

I have cloned the latest repo and am trying to build it, but am having multiple errors.  I reduced the env to build only for 
default_envs  = esp32doit-devkit-v1

When I build the project I get this error first:

`In file included from src/PedalinoMini.cpp:53:
src/Pedalino.h:418:6: error: 'vector' in namespace 'std' does not name a template type        
 std::vector<byte> ultrasonicEcho(PEDALS);         // Echo pins
      ^~~~~~
src/Pedalino.h:418:1: note: 'std::vector' is defined in header '<vector>'; did you forget to '#include <vector>'?
`

I did add
`#include <vector>`
 to pedalino.h and that resolved this error but now I get many more errors, the first one is:

`In file included from src/PedalinoMini.cpp:73:
src/WifiConnect.h: In function 'bool wps_config()':
src/WifiConnect.h:481:7: error: 'struct esp_wps_config_t' has no member named 'crypto_funcs'  
   WPS.crypto_funcs = &g_wifi_default_wps_crypto_funcs;
       ^~~~~~~~~~~~
src/WifiConnect.h:481:23: error: 'g_wifi_default_wps_crypto_funcs' was not declared in this scope
   WPS.crypto_funcs = &g_wifi_default_wps_crypto_funcs;
 `

I feel like I must be missing something simple?  I tried it on a completely different machine with all fresh installation of everything and I get the exact same thing.

**Comment by alf45tar** [comment, marked as answer]
The reason is the new version 4.1.0 of the espressif32 platform. Modified platformio.ini to use version 3.5.0.

**Comment by MichSenpai** [comment]
How we must modify platformio.ini?

**Comment by alf45tar** [comment]
The bug is fixed in sponsor version. However the hint is to change the line as per below

`platform 	           = espressif32@3.5.0`
